Kernel/FileSystem/Ext2Base.h

```diff
@@ -0,0 +1,307 @@
+    // know about, it should refuse to mount the filesystem.
+    //
+    // e2fsck's requirements are more strict; if it doesn't know
+    // about a feature in either the compatible or incompatible
+    // feature set, it must abort and not try to meddle with
+    // things it doesn't understand...
+
+    u32 first_inode_num;        // First non-reserved inode
+    u16 inode_size_num;         // size of inode structure
+    u16 num_block_group;        // block group # of this superblock
+    u32 compatible_features;    // compatible feature set
+    u32 incompatible_features;  // incompatible feature set
+    u32 ro_compatible_features; // readonly-compatible feature set
+    u8 volume_uuid[16];         // 128-bit uuid for volume
+    char volume_name[16];       // volume name
+    char last_mounted_dir[64];  // directory where last mounted
+    u32 algorithm_usage_bitmap; // For compression
+
+    // Performance hints.  Directory preallocation should only
+    // happen if the EXT2_FEATURE_COMPAT_DIR_PREALLOC flag is on.
+    u8 prealloc_blocks;      // Number of blocks to try to preallocate
+    u8 prealloc_dir_blocks;  // Number to preallocate for dirs
+    u16 reserved_gdt_blocks; // Per group table for online growth
+
+    // Journaling support valid if EXT2_FEATURE_COMPAT_HAS_JOURNAL is set.
+    u8 journal_sb_uuid[16];        // uuid of journal superblock
+    u32 journal_inode_num;         // inode number of journal file
+    u32 journal_device;            // device number of journal file
+    u32 last_orphan;               // start of list of inodes to delete
+    u32 htree_hash_seed[4];        // HTREE hash seed
+    u8 default_hash_version;       // Default hash version to use
+    u8 journal_backup_type;        // Default type of journal backup
+    u16 group_descriptor_size_num; // Group desc. size if INCOMPAT_64BIT is set
+    u32 default_mount_options;
+    u32 first_meta_block_group;   // First metablock group
+    u32 fs_create_time;           // When the filesystem was created
+    u32 journal_inode_backup[17]; // Backup of the journal inode
+    u32 num_blocks_high;          // Blocks count high 32bits
+    u32 num_reserved_blocks_high; // Reserved blocks count high 32 bits
+    u32 num_free_blocks_high;     // Free blocks count
+    u16 min_extra_inode_size;     // All inodes have at least # bytes
+    u16 want_extra_inode_size;    // New inodes should reserve # bytes
+    u32 misc_flags;               // Miscellaneous flags
+    u16 raid_stride;              // RAID stride
+    u16 mmp_interval;             // # seconds to wait in MMP checking
+    u64 mmp_block;                // Block for multi-mount protection
+    u32 raid_stripe_width;        // blocks on all data disks (N*stride)
+    u8 log_groups_per_flex;       // FLEX_BG group size
+    u8 reserved_char_pad;
+    u16 reserved_pad;  // Padding to next 32bits
+    u32 reserved[162]; // Padding to the end of the block
```

Comment:
Pretty sure `structs` can have members without names. Maybe we should use that here instead of coming up with new variations of "reserved". Same for other appearances of "reserved".

Reply:
I don't think that works. Do you mind giving me a reference/specification/something where that's allowed?

Reply:
Ah, that seems to only work for bit-fields specifically. Don't mind me then.